core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -227,6 +227,14 @@ class BaseUnit : RulesetObject(), INonPerpetualConstruction {
         return true
     }
 
+    // This returns the name of the unit this tech upgrades this unit to,
+    // or null if there is no automatic upgrade at that tech.
+    fun automaticallyUpgradedInProductionToUnitByTech(techName: String): String? {
+        if (obsoleteTech != null && obsoleteTech == techName)
```

Comment:
It's weird that you have 'techs that obsolete this' but you're still using the obsoleteTech to perform this function, shouldn't we *not* be using that anymore?

Reply:
My point here still stands